File: frontend/src/utils/declination.ts
// Склонение «товар / товара / товаров» и похожих фраз в UI
export const declinationWord = (n: number, one: string, two: string, many: string): string => {
    const lastTwoDigits = Math.abs(n) % 100

    if (lastTwoDigits >= 11 && lastTwoDigits <= 14) {
        return many
    }

    switch (Math.abs(n) % 10) {
        case 1:
            return one
        case 2:
        case 3:
        case 4:
            return two
        default:
            return many
    }
  }

File: backend/utils/declination.go
package utils

// Склонение слов
func DeclinationWord(n int, one, two, many string) string {
    absN := n
    if absN < 0 {
        absN = -absN
    }

    lastTwoDigits := absN % 100

    if lastTwoDigits >= 11 && lastTwoDigits <= 14 {
        return many
    }

    switch absN % 10 {
    case 1:
        return one
    case 2, 3, 4:
        return two
    default:
        return many
    }
}